### PR TITLE
Mouse wheel support for X11-less linux distros (openelec)

### PR DESCRIPTION
**Pull request by rkraevskiy**

### CODE REVIEW

**Comment by rkraevskiy** [comment]
fixed

**Comment by fritsch** [comment]
Please squash the two commits. Thanks much in advance.

**Comment by rkraevskiy** [comment]
done

**Comment by popcornmix** [comment]
This will be in tonight's Milhouse build (as a few users have been requesting it).
Initial testing looked good from a functional perspective.

File: xbmc/input/linux/LinuxInputDevices.cpp [hunk 1]
```diff
@@ -556,27 +558,37 @@ bool CLinuxInputDevice::KeyEvent(const struct input_event& levt, XBMC_Event& dev
...
-    devt.motion.xrel = 0;
-    devt.motion.yrel = levt.value;
+    devt[0].motion.xrel = 0;
+    devt[0].motion.yrel = levt.value;
+    motion = true;
     break;
-
-  case REL_Z:
   case REL_WHEEL:
+    if (levt.value != 0)
```
Comment (Paxxi) [inline xbmc/input/linux/LinuxInputDevices.cpp]: Can be replaced with wheel = levt.value != 0;

File: xbmc/input/linux/LinuxInputDevices.cpp [hunk 2]
```diff
@@ -691,61 +722,75 @@ XBMC_Event CLinuxInputDevice::ReadEvent()
...
-      if (m_hasLeds && (m_keyMods != m_lastKeyMods))
-      {
-        SetLed(LED_NUML, m_keyMods & XBMCKMOD_NUM);
-        SetLed(LED_CAPSL, m_keyMods & XBMCKMOD_CAPS);
-        m_lastKeyMods = m_keyMods;
-      }
+        // sanity check if we realy read the event
+        if(readlen != sizeof(levt))
+        {
+           printf("CLinuxInputDevice: read error : %s\n", strerror(errno));
```
Comment (Paxxi) [inline xbmc/input/linux/LinuxInputDevices.cpp]: This should use CLog and not printf

**Comment by Paxxi** [comment]
There's no reason to change the method signatures, if two events are produced just stick the overflow in m_queue in the translation method

File: xbmc/input/linux/LinuxInputDevices.cpp (hunk 2 above)
Comment (rkraevskiy) [inline xbmc/input/linux/LinuxInputDevices.cpp]: In the original code there  was printf instead CLog.

**Comment by rkraevskiy** [comment]
From my POV putting  the events to a queue inside ReadEvent more clear then putting same events somewhere inside translate method call tree.

File: xbmc/input/linux/LinuxInputDevices.cpp (hunk 1 above)
Comment (rkraevskiy) [inline xbmc/input/linux/LinuxInputDevices.cpp]: I propose to leave it as is.

**Comment by Paxxi** [comment]
I understand that but it messes up the rest of the code having a bunch of [0] everywhere that's not really needed

**Comment by rkraevskiy** [comment]
From my POV putting  the events to a queue inside ReadEvent more clear then putting same events somewhere inside translate method call tree.

**Comment by Paxxi** [comment]
Old code shouldn't have been using printf there either but was likely some leftover that we had missed. Please update it to CLog and you have my ok.

Should probably be good to merge once @MilhouseVH build has been tested.

**Comment by popcornmix** [comment]
My testing and reports on forum are positive. Any objections to this going in?
jenkins build this please

**Pull request merged**